### ISSUE TITLE
Errors when upgrading "@xyflow/react"

### ISSUE THREAD

**Issue by sroussey** [clarkmcc/ngraph#46]
Various issues when upgrading "@xyflow/react" to beta 10, 11, or 12. Looks like type stuff changes, particularly in the hooks.

**Comment by clarkmcc** [comment]
Makes sense. I'm happy to accept a PR, otherwise I'll hold off until the official release before I fully migrate the library. Release date should be very soon https://github.com/xyflow/xyflow/discussions/3764#discussioncomment-8636409

**Comment by sroussey** [comment]
I made changes, I'll work on a PR for it
